src/backend/web/validation.js

```diff
@@ -50,9 +51,33 @@ function validateNewFeed() {
     }
   };
 }
+// Rules for Validation of query params for the /posts route
+const validatePostsQueryRules = [
+  query('per_page', 'per_page needs to be empty or a integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+  query('page', 'page needs to be empty or an integer').custom(
+    (value) => !value || Number.isInteger(+value)
+  ),
+];
+// create the middle ware for the /posts route with error reporting
+const validatePostsQuery = () => {
+  return async (req, res, next) => {
```

Comment:
This code is identical to `validateQuery`, with the exception  of the rules.  Can you refactor this into a function we call in both places?  Something like this (untested):

```js
const validate = (rules) => async (req, res, next) => {
    await Promise.all(rules().map((rule) => rule.run(req)));
    const errors = validationResult(req);
    if (errors.isEmpty()) {
      return next();
    }
    return res.status(400).json({
      errors: errors.array(),
    });
  };

const validateQuery = () => validate(validatePostsQueryRules);
```

Reply:
Ya, I had that idea before Ill do that now.